src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -1043,6 +1073,273 @@ public static void Xml_DeserializeEmptyDateTimeOffsetType()
         }
     }
 
+    [Theory]
+    [InlineData("0001-1-1")]
+    [InlineData("2002-6-17")]
+    [InlineData("2345-12-1")]
+    public static void Xml_DateOnlyAsRoot(string dateString)
+    {
+        var doObj = DateOnly.Parse(dateString);
+        var result = SerializeAndDeserialize<DateOnly>(doObj, WithXmlHeader($"""
+            <dateOnly>{FormatDateString(doObj)}</dateOnly>
+            """));
+        Assert.StrictEqual(doObj, result);
+    }
+
+    [Theory]
+    [InlineData("1-1-1")]
+    [InlineData("0079-08-24+02:00")]    // Vesuvius
+    [InlineData("1912-04-15Z")]         // Titanic - 11/14 11:40 ships time
+    [InlineData("1917-12-6")]           // Halifax
+    [InlineData("1937-5-06")]           // Hindenburg
+    [InlineData("98-01-01")]            // Rose Bowl
+    public static void Xml_DateOnlyParseErrors(string badDateString)
+    {
+        var badXml = WithXmlHeader($"""
+            <dateOnly>{badDateString}</dateOnly>
+            """);
+        Assert.Throws<InvalidOperationException>(() => DeserializeFromXmlString<DateOnly>(badXml));
+    }
+
+    [Theory]
+    [InlineData("20:17:40")]  // The Eagle has landed
+    [InlineData("10:35 AM")]  // First in flight
+    [InlineData("10:45 PM")]  // Tear down this wall
+    public static void Xml_TimeOnlyAsRoot(string timeString)
+    {
+        var toObj = TimeOnly.Parse(timeString);
+        var result = SerializeAndDeserialize<TimeOnly>(toObj, WithXmlHeader($"""
+            <timeOnly>{FormatTimeString(toObj)}</timeOnly>
+            """));
+        Assert.StrictEqual(toObj, result);
+    }
+
+    [Theory]
+    [InlineData("07:25:13.45-04:00", true, "07:25:13.45")]  // Oh the humanity
+    [InlineData("02:38:23.40Z", true, "02:38:23.4")]        // My heart will go on
+    [InlineData("7:48:07", false)]        // Will live in infamy
+    [InlineData("08:32 AM", false)]       // Helen errupts
+    public static void Xml_TimeOnlyParseErrors(string timeString, bool succeedsWithCompat, string expected = "")
+    {
+        const string switchName = "Switch.System.Xml.AllowXsdTimeToTimeOnlyWithOffsetLoss";
+        AppContext.TryGetSwitch(switchName, out bool originalEnabled);
+
+        // Try straigt up
+        var xml = WithXmlHeader($"<timeOnly>{timeString}</timeOnly>");
+        TimeOnly result = default;
+        var ex = Record.Exception(() =>
+        {
+            result = DeserializeFromXmlString<TimeOnly>(xml);
+        });
+        Assert.NotNull(ex);
+        Assert.IsType<InvalidOperationException>(ex);
+
```

Comment:
Can replace this with:
```c#
Assert.Throws<InvalidOperationException>(() =>
{
    result = DeserializeFromXmlString<TimeOnly>(xml);
});
```
It checks everything all at once.

Reply:
I think using this format was a function of the rest of the test method where I need to optionally log the exception and check it based on the current configuration.

Reply:
You left the original code there, so you're doing the same thing twice. If you want to stick to the Record.Exception call for consistency, that's fine. By Assert.Throws will assert if it doesn't throw, which covers the Assert.NotNull(ex) check, and specifying the exception type also covers the Assert.IsType. Also, Assert.Throws returns the actual exception so you could to `var ex = Assert.Throws....` and keep your not null and type checks.